### ISSUE TITLE
Suggestion: use `:{type}` to omit formal parameter when declare function interface or something else

### ISSUE THREAD

**Issue by Treri** [Microsoft/TypeScript#10984]
When we declare a function interface, omit formal parameter. Because the param name is not important in interface, just the param type does.

So, we can make below code

```ts
interface func{
    (paramA: typeA, paramB: typeB): typeC;
}
```

to this form

```ts
interface func{
    (:typeA, :typeB): typeC
}
```

**Comment by kitsonk** [comment]
The formal parameter name is often useful for intellisense when using semantically meaningful parameter names, is it not?

**Comment by RyanCavanaugh** [comment]
I mean, you can't stop someone from *not* providing a useful name. `p0, p1, p2`, etc.

**Comment by Treri** [comment]
I mean, TypeScript can support this form but it's optional. So someone can emit the formal parameter name. Because sometime we don't know the correct parameter name until we implement it

**Comment by yortus** [comment]
There are situations [like this](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9b111fde729ec9f105c01930f98fab212e0cb21e/bluebird/bluebird.d.ts#L388-L394) where omitting some parameter names would be a small mercy. But that particular pattern will hopefully one day be addressed better by something like #5453.

In general there *are* cases where the intent of declaring a *function type* is purely to describe a mapping from certain parameter type(s) to a certain return type, and giving names to the parameters is mostly meaningless boilerplate. You can see some of these in [ramda.d.ts](https://github.com/donnut/typescript-ramda/blob/master/ramda.d.ts).

Basically any situation where you'd be temped to name things `arg1`, `arg2` and so on would be candidates for this proposal. The compiler would probably internally name them `arg1`, `arg2` etc for intellisense anyway.

**Comment by RyanCavanaugh** [comment]
We'd really like people to try hard to come up with parameter names, and this syntax looks really weird on its own (is it a weird kind of parameter name? a destructuring? something else?) so we'd prefer to keep names mandatory at this point.
